Library/Homebrew/requirements.rb

```diff
@@ -134,7 +146,8 @@ def initialize(tags)
   def java_version
     args = %w[/usr/libexec/java_home --failfast]
     args << "--version" << "#{@version}+" if @version
-    quiet_system(*args)
+    @java_home = `#{args * " "}`.chomp
```

Comment:
Mmm, backticks here allow you to run any arbritrary code by specificying it as the Java version e.g. `depends_on :java => "1.7+; rm -rf ~"`. I realise we probably have a bunch of other places where that's possible but would be nice to work around it if we can.